### PR TITLE
https-dns-proxy: bump the version

### PR DESCRIPTION
**Pull request by Nowaker**
I've no idea what I should set `PKG_MIRROR_HASH` to. We need it documented in https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md. EDIT: I used what the CI told me the hash was... Not ideal.

Maintainer: @stangri 
Compile tested: @Nowaker
Run tested: @Nowaker, works on Turris Omnia:

```
# cat /etc/openwrt_release 
DISTRIB_ID='TurrisOS'
DISTRIB_RELEASE='5.1.1'
DISTRIB_REVISION='86b4402'
DISTRIB_TARGET='mvebu/cortexa9'
DISTRIB_ARCH='arm_cortex-a9_vfpv3-d16'
DISTRIB_DESCRIPTION='TurrisOS 5.1.1 86b44028e2123a199d9ff9cfc1585696bed7af53'
DISTRIB_TAINTS='busybox'
```

Description: Bumps to the latest version in upstream. The newest version most likely fixes a problem described in https://github.com/aarond10/https_dns_proxy/issues/92 in commit https://github.com/aarond10/https_dns_proxy/commit/7a825c1f4e5f26915845823fb50dba1bb601a241.

### CODE REVIEW

**Comment by stangri** [comment]
Thank you for bring it up. I have the correct Makefile at https://github.com/stangri/openwrt_packages/blob/master/https-dns-proxy/Makefile. I'll need to test this on both 18.06 and 19.07 before creating PR.

**Comment by Nowaker** [comment]
Thanks @stangri. I'm currently trying to get the toolchain for Turris Omnia running so if I'm successful, you'll have a second person to give it a test.

**Comment by stangri** [comment]
> Thanks @stangri. I'm currently trying to get the toolchain for Turris Omnia running so if I'm successful, you'll have a second person to give it a test.

Omnia is mvebu, right? If you're on 19.07, you can try installing this: https://dev.melmac.net/openwrt-repo/https-dns-proxy_2020-08-21-1_arm_cortex-a9_vfpv3-d16.ipk

**Comment by Nowaker** [comment]
> Omnia is mvebu, right?

Not sure. `uname -a` says `armv7l`.

I tried installing your package but `Cannot satisfy the following dependencies for https-dns-proxy: libcurl4`. What I see in `opkg list` is `libcurl - 7.66.0-3 - A client-side URL transfer library`.

**Comment by Nowaker** [comment]
I think I got the toolchain running, and my own version of `https-dns-proxy` is now compiling.

**Comment by Nowaker** [comment]
I can confirm `https-dns-proxy` works just fine after having been compiled with the Makefile in this PR.

**Comment by stangri** [comment]
> I can confirm `https-dns-proxy` works just fine after having been compiled with the Makefile in this PR. PR description updated to confirm it's compiled and tested.

Weird, I get different PKG_MIRROR_HASH. I mean it should probably still compile but would have to re-download the new xz on each compile.

File: net/https-dns-proxy/Makefile [hunk 1]
```diff
@@ -1,14 +1,14 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=https-dns-proxy
-PKG_VERSION:=2020-04-09
-PKG_RELEASE=3
+PKG_VERSION:=2020-09-24
+PKG_RELEASE=4
```
Comment (BKPepe) [inline net/https-dns-proxy/Makefile:5]:
```suggestion
PKG_RELEASE=1
```

**Comment by BKPepe** [comment]
> > Thanks @stangri. I'm currently trying to get the toolchain for Turris Omnia running so if I'm successful, you'll have a second person to give it a test.
> 
> Omnia is mvebu, right?

Yes, that's correct. Also, the package which you linked should work on Turris Omnia.

**Comment by Nowaker** [comment]
> Weird, I get different PKG_MIRROR_HASH

@stangri The CI agrees with my checksum so I guess it's a good one.

@BKPepe Thanks for explaining the process. I also applied your suggestion. I understand after this gets accepted to master I should create a cherry-pick PR for the `openwrt-19.07`, correct?

**Comment by stangri** [comment]
@Nowaker thanks for persevering with this. 

I insist on `PKG_VERSION:=2020-08-21` tho. As there are no versions for principal package, the release date is the closest thing to a version it has.

As far as I can see, you're running/building for non-vanilla OpenWrt, I don't know what the practices are in that regard, maybe @BKPepe can fill us in.

I've already tested the update on 18.06 and I will be able to test on 19.07 over the weekend. I can submit a properly formatted PR then.

**Comment by Nowaker** [comment]
> @Nowaker thanks for persevering with this.
> 
> I insist on `PKG_VERSION:=2020-08-21` tho. As there are no versions for principal package, the release date is the closest thing to a version it has.

OK. I would vaguely interpret [the guideline](https://openwrt.org/docs/guide-developer/package-policies#versioning) as   `PKG_SOURCE_DATE = PKG_VERSION` for git repos. I've updated the PR, and I'm going to edit the policy as well.

> As far as I can see, you're running/building for non-vanilla OpenWrt, I don't know what the practices are in that regard, maybe @BKPepe can fill us in.

I got all working so we're good.

**Comment by Nowaker** [comment]
@stangri Actually... The guideline reads:

> the ''PKG_SOURCE_DATE'' and ''PKG_SOURCE_VERSION'' variables should be used **instead** [of PKG_VERSION]

Which means we should not define `PKG_VERSION` at all... Right?

This is what happens when a guideline is written as prose, rather than being a very strict technical document like Arch's [PKGBUILD](https://wiki.archlinux.org/index.php/PKGBUILD).

> Weird, I get different PKG_MIRROR_HASH

Well, the hash changes when you change PKG_VERSION. That's extremely counterintuitive and seems to contradict the point of having checksums.

**Comment by neheb** [comment]
PKG_VERSION gets autogenerated from PKG_SOURCE_VERSION. See https://github.com/openwrt/openwrt/blob/master/include/download.mk#L14

Using PKG_SOURCE_VERSION is usually simpler.

**Comment by stangri** [comment]
Superseded by https://github.com/openwrt/packages/pull/13512.

@neheb -- I have additional Makefile questions, I'll try to PM/email you separately.

**Comment by Nowaker** [comment]
Not really sure what the point of submitting a second PR is when mine is functionally the same, and everything has been tested.

**Comment by BKPepe** [comment]
I don't understand that, either.  Because those changes are the same and one point of view can be that the maintainer took your changes and commit them himself as he wants to know how many people view his README in this repo, which doesn't seem to be alright. I'm sorry for that.

**Comment by Nowaker** [comment]
I don't mind the README, as it's good intel for the maintainer (he can see people are actually using his package and be more motivated to keep it up to date). That said, getting my changes merged would be a personal reward to myself for being a first-time contributor to OpenWRT packages. :trophy:

**Comment by stangri** [comment]
> Not really sure what the point of submitting a second PR is when mine is functionally the same, and everything has been tested.

Last I checked, the PKG_MIRROR_HASH was wrong (I see now that at some point you have copied the value from the Makefile I linked in my first reply) and this was compiled and tested against non-vanilla OpenWrt, as well as having formatting issues, so IMHO had low probability of being accepted.

**Comment by Nowaker** [comment]
> Last I checked, the PKG_MIRROR_HASH was wrong (I see now that at some point you have copied the value from the Makefile I linked in my first reply)

I copied it from the CI.

> and this was compiled and tested against non-vanilla OpenWrt, 

...and you could test it as a team effort.

> as well as having formatting issues

Please check the diff. What formatting issues exactly?

**Comment by stangri** [comment]
> I don't understand that, either. Because those changes are the same and one point of view can be that the maintainer took your changes and commit them himself as he wants to know how many people view his README in this repo, which doesn't seem to be alright. I'm sorry for that.

An informed point of view is that multiple times maintainer mentioned that committer used incorrect PKG_MIRROR_HASH and also provided the Makefile with the correct value, to which the response was "it's from CI, it's all good". The maintainer also brought up the requirements for compile-tested and run-tested in the description which are to this date have not been fixed.

Maybe I haven't used the github review to its full potential or didn't make some things clear enough, but the hostility is totally uncalled for.

**Comment by stangri** [comment]
> > Last I checked, the PKG_MIRROR_HASH was wrong (I see now that at some point you have copied the value from the Makefile I linked in my first reply)
> 
> I copied it from the CI.
> 
> > and this was compiled and tested against non-vanilla OpenWrt,
> 
> ...and you could test it as a team effort.

If you ended up using the Makefile I posted 6 days ago, that has been compile and run-tested as per my PR. Now we're half-way there to a successful PR.

> > as well as having formatting issues
> 
> Please check the diff. What formatting issues exactly?

Sorry, I should have been clearer -- not the formatting, the description, like I said before there are formal requirements for compile-tested and run-tested information there. You can just copy information from my PR.

**Comment by stangri** [comment]
> I don't understand that, either. Because those changes are the same and one point of view can be that the maintainer took your changes and commit them himself as he wants to know how many people view his README in this repo, which doesn't seem to be alright. I'm sorry for that.

Do you want to apologize for insinuating that I stole the code from the committer when it's evident I have posted the correct Makefile for the committer to use and not the other way around?

**Comment by neheb** [comment]
https://github.com/openwrt/packages/pull/13512 was merged. Closing.

**Pull request closed without merging**